ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -192,20 +203,28 @@ module Json = struct
         `O [("import", `Bool true); ("items", `A items)]
 
   let xenapi objs =
-    let enum_set = ref EnumSet.empty in
-    let enum_dedup enums1 enums2 =
-      let set =
-        EnumSet.of_list enums1 |> EnumSet.union (EnumSet.of_list enums2)
+    let enums_acc = ref StringMap.empty in
+    let erase_existed enums =
```

Comment:
I don't understand what this code is meant to do. It looks like when an enum is attached to an object, it can't be used for others, but that doesn't seem like the correct thing to do.

Ideally, enums would be defined independently and be references correctly in all objects that need it.

Reply:
This a example for generated enum type:
```go
type HostDisplay string

const (
	// This host is outputting its console to a physical display device
	HostDisplayEnabled HostDisplay = "enabled"
	// The host will stop outputting its console to a physical display device on next boot
	HostDisplayDisableOnReboot HostDisplay = "disable_on_reboot"
	// This host is not outputting its console to a physical display device
	HostDisplayDisabled HostDisplay = "disabled"
	// The host will start outputting its console to a physical display device on next boot
	HostDisplayEnableOnReboot HostDisplay = "enable_on_reboot"
)
``` 
When define a  `const` variable in one Go file and then define another `const` variable with the same name in a different file, the Go linter would give a warning. This warning occurs because the Go linter is designed to enforce good coding practices and maintainability by flagging potential issues such as redefining constants with the same name.

It's a nice way that enums are defined in a file. Tomorrow I'll check with Sola and refactor it.

Reply:
How about to put all Enum types into one Go file? Since actually one Enum can be referenced by all modules.